Services - Three-column boxed pricing table

noruzzamans opened the pull request:
### Testing Instructions
Create a new page.
Add the pattern.
Confirm that it looks like the designs (spacing/colors/etc.).

Closes: https://github.com/WordPress/twentytwentyfive/issues/76
[Figma](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1809&m=dev)

shail-mehta commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,141 @@
+ */
+
+?>
+
```
Could you please removed this empty line before `<!-- wp:group -->` ?

shail-mehta commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,141 @@
+
+?>
+
+<!-- wp:group {"metadata":{"name":"Services - Three-column boxed pricing table","categories":["services"],"patternName":"twentytwentyfive/services-three-column-boxed-pricing-table"},"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50"},"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
```
Could you please remove metadata ?

shail-mehta commented:
1. We should use presets instead of hardcoding values. Padding should be adjusted according to the spacing presets.
**Ref**: https://github.com/WordPress/twentytwentyfive/issues/34

2. We should use font sizes according to the default presets.
**Example:** if font size is 16px: need to use `has-small-font-size`

rejaulalomkhan commented:
Thanks for the PR @noruzzamans. 
I have tested your design. Your pattern is working fine. but some improvements (`size-related presets` ) will make your design perfect.

noruzzamans commented:
> 1. We should use presets instead of hardcoding values. Padding should be adjusted according to the spacing presets.
>    **Ref**: [Add width, block gap and spacing presets #34](https://github.com/WordPress/twentytwentyfive/issues/34)
> 2. We should use font sizes according to the default presets.
>    **Example:** if font size is 16px: need to use `has-small-font-size`

Thanks for reviewing my pattern. You suggested that I use font sizes according to the default presets, but the Figma design has custom sizes, such as 12px for the "Become a member" button, 38px and 12px for the pricing font, and 18px for the "Join" button. How can I adjust these to fit the default presets?

carolinan commented:
>Thanks for reviewing my pattern. You suggested that I use font sizes according to the default presets, but the Figma design has custom sizes, such as 12px for the "Become a member" button, 38px and 12px for the pricing font, and 18px for the "Join" button. How can I adjust these to fit the default presets?

Select the closest matching size in the theme.json file:
12px is replaced with the "small" font size.
Use the default font size for "Join" buttons. Do not add a font size for these.
Use x-large for the price, the numbers.

Remove the contrast text color, because this is the default color.
Remove the font family, because it is the default family.
Remove the padding on the Join buttons, and use the default.

carolinan commented:
Please check if you can make the heading "Choose your membership" and the small button wrap to separate lines on small screens.

noruzzamans commented:
> > Thanks for reviewing my pattern. You suggested that I use font sizes according to the default presets, but the Figma design has custom sizes, such as 12px for the "Become a member" button, 38px and 12px for the pricing font, and 18px for the "Join" button. How can I adjust these to fit the default presets?
> 
> Select the closest matching size in the theme.json file: 12px is replaced with the "small" font size. Use the default font size for "Join" buttons. Do not add a font size for these. Use x-large for the price, the numbers.
> 
> Remove the contrast text color, because this is the default color. Remove the font family, because it is the default family. Remove the padding on the Join buttons, and use the default.

I have followed all your instructions and made the necessary updates. However, I did not update the theme.json. The "small" font size currently represents 14px, and updating it to 12px would apply the change globally, which could potentially disrupt the designs of other contributors. Instead, I used the closest matching size to 12px, and the design still looks good. Could you please confirm if this approach is acceptable? If not, we could consider adding an extra font size named 'X-Small' for 12px. Please review my update pattern and let me know if any further changes are needed. Thank you.

noruzzamans commented:
> Please check if you can make the heading "Choose your membership" and the small button wrap to separate lines on small screens.

I have built this with a two-column layout, using a button in the right column positioned to the right. When using the default column setting "Stack on mobile" for a separated line, the buttons appear on the right side on mobile, which doesn't look good. I do not see an option in the settings panel to adjust the button's alignment to the left on mobile. Could you please let me know how I can achieve this? If you have any suggestions, please let me know. Thank you.

carolinan commented:
> However, I did not update the theme.json. The "small" font size currently represents 14px, and updating it to 12px would apply the change globall

Yes, that is correct,  I did not mean for you to update theme.json, but use the closest font size that is in theme.json.

It has not been possible to keep the Figma file updated with the necessary changes, including the removal of some font sizes and renaming of colors. I understand that if this is frustrating, but there are several attributes that needs to be removed from the markup that is submitted, and replaced with the existing presets.

carolinan commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,140 @@
+<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:40%"><!-- wp:buttons {"align":"wide","layout":{"type":"flex","justifyContent":"right","orientation":"horizontal","flexWrap":"nowrap"}} -->
+<div class="wp-block-buttons alignwide"><!-- wp:button {"className":"is-style-outline","style":{"border":{"style":"solid","width":"1px"},"typography":{"fontStyle":"normal","fontWeight":"400"}},"fontSize":"small","borderColor":"contrast"} -->
+<div class="wp-block-button has-custom-font-size is-style-outline has-small-font-size" style="font-style:normal;font-weight:400"><a class="wp-block-button__link has-border-color has-contrast-border-color wp-element-button" style="border-style:solid;border-width:1px" rel="#">Become a member</a></div>
+<!-- /wp:button --></div>
```
I think the use of button here is smart, but I think before the theme is finalized, this will be a block style variation on the paragraph block, it is still being discussed in other pull requests how to solve this. So it might take some additional time before this PR can be merged.

noruzzamans replied:
Thanks for the feedback; I'll follow the discussion and update the implementation once a decision is made.

carolinan replied:
Hi, the pill style variation for the paragraph block has been merged.
It needs to be placed inside a row, or it will be too wide, so I suggest trying to replace the columns with a row.
Please test it, you may need to make further changes than only copy and pasting this. 😃 

```
<!-- wp:group {"align":"wide","layout":{"type":"flex","flexWrap":"wrap","justifyContent":"space-between"}} -->
<div class="wp-block-group alignwide"><!-- wp:heading -->
<h2 class="wp-block-heading">Choose your membership</h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"className":"is-style-pill"} -->
<p class="is-style-pill">Become a member</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->
```

noruzzamans replied:
Thanks for your guidance. I have resolved the pill style issues. Please review and let me know if any changes are needed

carolinan commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,140 @@
+/**
+ * Title: Services - Three-column boxed pricing table
+ * Slug: twentytwentyfive/services-three-column-boxed-pricing-table
+ * Categories: call-to-action, banner, featured, services, pricing
```
The pricing category does no exist, so please remove it.

noruzzamans replied:
@carolinan I have removed the pricing category

carolinan commented:
Hi

The pattern needs to be wrapped inside a full width group block. Right now, there is no way for the user to add a background color or change the spacing around the whole pattern.
Set the top and bottom margin of this new group to 0, so that there is no extra spacing between this pattern and other full width patterns.
Then set top and bottom padding to spacing preset 50: regular. (This will match the other service pattern that is already in the theme).

You will need to adjust the existing top and bottom spacing on the inner group and columns as well so that the space is not doubled.

---

Why is the group block inside the columns of the first, outer columns block needed? I'd like to understand why this was chosen instead of placing the spacing and border on the column itself.

----

I don't understand why a columns block is used as a container for the single buttons? There is only one column.
Instead of hard-coding 21px for the spacing between the text and the button, please use the closest spacing preset that is available in theme.json. That way the value is easier to adjust in the future.

----
The first heading, "Choose your membership" is an `H2`, and the headings inside the columns are `H4`.
The heading level must not be skipped, it can make it more difficult for the reader to navigate the content.
Please change the headings that are inside the 3 columns to `H3`, but make sure to keep the correct font size, font weight etc.

----
I see that both the column and the inner text elements have the typography settings applied.:

```
<!-- wp:column {"style":{"typography":{"fontSize":"22px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"}}} -->
<!-- wp:heading {"level":4,"style":{"typography":{"lineHeight":"1.8"}},"fontSize":"large"} -->
<!-- wp:paragraph {"style":{"typography":{"fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"},"spacing":{"margin":{"top":"0","bottom":"0","left":"0","right":"0"}}},"fontSize":"small"} -->
```

Do the text blocks not inherit the typography settings from the column? If they don't, you can remove it from the column.
Having said that, the 22px value must be replaced with a preset.

----

The borders have a value of `11111133;` please use the color preset `opacity-20`.

noruzzamans commented:
> Hi
> 
> The pattern needs to be wrapped inside a full width group block. Right now, there is no way for the user to add a background color or change the spacing around the whole pattern. Set the top and bottom margin of this new group to 0, so that there is no extra spacing between this pattern and other full width patterns. Then set top and bottom padding to spacing preset 50: regular. (This will match the other service pattern that is already in the theme).
> 
> You will need to adjust the existing top and bottom spacing on the inner group and columns as well so that the space is not doubled.
> 
> Why is the group block inside the columns of the first, outer columns block needed? I'd like to understand why this was chosen instead of placing the spacing and border on the column itself.
> 
> I don't understand why a columns block is used as a container for the single buttons? There is only one column. Instead of hard-coding 21px for the spacing between the text and the button, please use the closest spacing preset that is available in theme.json. That way the value is easier to adjust in the future.
> 
> The first heading, "Choose your membership" is an `H2`, and the headings inside the columns are `H4`. The heading level must not be skipped, it can make it more difficult for the reader to navigate the content. Please change the headings that are inside the 3 columns to `H3`, but make sure to keep the correct font size, font weight etc.
> 
> I see that both the column and the inner text elements have the typography settings applied.:
> 
> ```
> <!-- wp:column {"style":{"typography":{"fontSize":"22px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"}}} -->
> <!-- wp:heading {"level":4,"style":{"typography":{"lineHeight":"1.8"}},"fontSize":"large"} -->
> <!-- wp:paragraph {"style":{"typography":{"fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"},"spacing":{"margin":{"top":"0","bottom":"0","left":"0","right":"0"}}},"fontSize":"small"} -->
> ```
> 
> Do the text blocks not inherit the typography settings from the column? If they don't, you can remove it from the column. Having said that, the 22px value must be replaced with a preset.
> 
> The borders have a value of `11111133;` please use the color preset `opacity-20`.

@carolinan I have added a full-width group block to resolve the issue with background color and spacing. The top and bottom margins have been set to 0, and the padding has been adjusted to the spacing preset '50: regular,' ensuring consistency with the other service patterns in the theme.

Regarding the group block within the outer columns, I used this structure because the columns block does not support border-radius. By placing the group block inside, I was able to apply the necessary border-radius, while still utilizing columns for content and layout.

Additionally, I've adjusted the top and bottom spacing on the inner group and columns to avoid doubling the space. The columns block was used to ensure layout flexibility and responsiveness.

For the single button, I have replaced the hardcoded 21px spacing between the text and button with the closest spacing preset from theme.json, making it easier to adjust in the future.

The heading "Choose your membership" is now set as an H2, while the headings within the columns have been updated to H3 to maintain proper hierarchy and improve accessibility. I ensured the correct font size, weight, and other styling attributes were retained after the change.

I also reviewed the typography settings. I removed the typography settings from the inner text elements.

For the borders, I used the color value #11111133 and applied the color preset with 20% opacity, ensuring it adheres to the theme's design system.

Please review and let me know if there are any further changes. Thanks

carolinan commented:
Hi, you need to use the latest Gutenberg plugin to use the latest features, this includes the border radius,
The changes made to Gutenberg before the cut-off date, which is next week, will be included in WordPress 6.7 unless something unforeseen happens.

carolinan commented:
> For the borders, I used the color value #11111133 and applied the color preset with 20% opacity, ensuring it adheres to the theme's design system.

I don't understand. Why is it not possible to use the preset called opacity 20?
By hardcoding the color here, the styles variations will not work because they won't override the `#11111133`

carolinan commented:
There needs to be a single empty line at the end of the PHP file.

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,122 @@
+
+<!-- wp:columns {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"0"}}}} -->
+<div class="wp-block-columns alignwide" style="margin-top:var(--wp--preset--spacing--70);margin-bottom:0"><!-- wp:column {"layout":{"type":"constrained","justifyContent":"center"}} -->
+<div class="wp-block-column"><!-- wp:group {"style":{"border":{"width":"1px","color":"#11111133","radius":"10px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}}},"layout":{"type":"default"}} -->
```
Can you please move the border styles to the columns and use the preset of opacity-20 for the color? the same for the other columns.

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,122 @@
+<!-- wp:columns {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"0"}}}} -->
+<div class="wp-block-columns alignwide" style="margin-top:var(--wp--preset--spacing--70);margin-bottom:0"><!-- wp:column {"layout":{"type":"constrained","justifyContent":"center"}} -->
+<div class="wp-block-column"><!-- wp:group {"style":{"border":{"width":"1px","color":"#11111133","radius":"10px"},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}}},"layout":{"type":"default"}} -->
+<div class="wp-block-group has-border-color" style="border-color:#11111133;border-width:1px;border-radius:10px;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:columns {"isStackedOnMobile":false,"style":{"spacing":{"blockGap":{"top":"0"},"margin":{"top":"0","bo ...
```
If we move the borders and spacing to the columns, we shouldn't need the groups. The same for the other columns.

noruzzamans replied:
> If we move the borders and spacing to the columns, we shouldn't need the groups. The same for the other columns.

@juanfra yes I have removed group.

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,122 @@
+<!-- /wp:column -->
+
+<!-- wp:column {"layout":{"type":"constrained"}} -->
+<div class="wp-block-column"><!-- wp:group {"style":{"border":{"width":"1px","color":"#11111133","radius":"10px"},"spacing":{"padding":{"top":"var:preset|spacing|40","right":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"30px"}}}} -->
```
Can we adjust the left spacing to use the preset? The same for the other ones.

noruzzamans replied:
@juanfra Yes, I have updated this part as well. Thank you so much for your help.

juanfra reviewed (commented):
Thank you for working on this. Can you please review the comments we left with Carolina? Thank you 🙏

noruzzamans commented:
> Hi, you need to use the latest Gutenberg plugin to use the latest features, this includes the border radius, The changes made to Gutenberg before the cut-off date, which is next week, will be included in WordPress 6.7 unless something unforeseen happens.

@carolinan Apologies, I previously did not use the Gutenberg plugin. Now, I have installed it and can see the border-radius feature in the column block, so I removed the extra group block. Thanks

noruzzamans commented:
> > For the borders, I used the color value #11111133 and applied the color preset with 20% opacity, ensuring it adheres to the theme's design system.
> 
> I don't understand. Why is it not possible to use the preset called opacity 20? By hardcoding the color here, the styles variations will not work because they won't override the `#11111133`

@carolinan I have resolved this issue by using a preset opacity of 20. Thanks

noruzzamans commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,122 @@
+
+<!-- wp:column {"style":{"typography":{"fontSize":"14px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"},"spacing":{"margin":{"top":"0","bottom":"0","left":"0","right":"0"}}}} -->
+<div class="wp-block-column" style="margin-top:0;margin-right:0;margin-bottom:0;margin-left:0;font-size:14px;font-style:normal;font-weight:400;line-height:1.8"><!-- wp:heading {"textAlign":"right","level":3} -->
+<h3 class="wp-block-heading has-text-align-right">0€</h3>
```
@juanfra I have updated the text-decoration to the line-through option.

noruzzamans commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,122 @@
+<!-- /wp:paragraph --></div>
+<!-- /wp:column -->
+
+<!-- wp:column {"style":{"typography":{"fontSize":"14px","fontStyle":"normal","fontWeight":"400","lineHeight":"1.8"},"spacing":{"margin":{"top":"0","bottom":"0","left":"0","right":"0"}}}} -->
```
> Please remove the hardcoded font-size from the columns, we should rely on the presets. The `14px` shouldn't be here. The same for the other columns.
> 
> I believe we shouldn't set the font style, and font weight here.

@juanfra I have removed the extra custom CSS

carolinan commented:
I believe that everything that was mentioned has been fixed, thank you.
The first column, the one with the free option, should not have the month below the price.

----

I have one question, I see that some of the top margins uses `0`, and others `0px` is there a reason for this?
Actually, when I search through the entire theme, there are 50 occurrences of  `"top":"0"` and 9 with` "top":"0px"`.
Now I am curios about why that is and if perhaps there is a change in Gutenberg that has applied the unit.

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,116 @@
+ */
+
+?>
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"},"margin":{"top":"0","bottom":"0"},"blockGap":"0px"}},"layout":{"type":"constrained"}} -->
```
```suggestion
<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|60","bottom":"var:preset|spacing|60","left":"var:preset|spacing|50","right":"var:preset|spacing|50"},"margin":{"top":"0","bottom":"0"},"blockGap":"0px"}},"layout":{"type":"constrained"}} -->
```

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,116 @@
+
+?>
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|50","left":"var:preset|spacing|50","right":"var:preset|spacing|50"},"margin":{"top":"0","bottom":"0"},"blockGap":"0px"}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group alignfull" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)"><!-- wp:group {"align":"wide","layout":{"type":"flex","flexWrap":"wrap","justifyContent":"space-between"}} -->
```
```suggestion
<div class="wp-block-group alignfull" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--60);padding-right:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--60);padding-left:var(--wp--preset--spacing--50)"><!-- wp:group {"align":"wide","layout":{"type":"flex","flexWrap":"wrap","justifyContent":"space-between"}} -->
```

juanfra reviewed (commented):
Thank you for working on this @noruzzamans!

Besides the comments Carolina has left, things are looking good to me. Once those are addressed, this one is good to be merged for me.

carolinan commented:
I did some quick testing: Let's leave the differences in the 0 margin as is, it does not create any issues.
If the user uses the spacing preset "none", I mean if they select the option in the preset option, it is set to 0. And if the user instead inputs the value 0 in the control, the unit is added.

juanfra commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,116 @@
+<h3 class="wp-block-heading has-text-align-right" style="text-decoration:line-through">0€</h3>
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"align":"right","style":{"spacing":{"margin":{"top":"0"}}},"fontSize":"small"} -->
+<p class="has-text-align-right has-small-font-size" style="margin-top:0">Month</p>
+<!-- /wp:paragraph --></div>
```
```suggestion
</div>
```

beafialho commented:
Thank you for sticking with it @noruzzamans! It's looking quite good, I have a few fine-tuning comments:

- The Columns block should have `Block Spacing: Regular` both vertically and horizontally.
- The paragraphs beneath "Free", "Single" and "Expert" should not have line height defined locally, and should be set to `Medium font size` instead
- Beneath the 0€ there should be no "month" word

noruzzamans commented:
@juanfra I have resolved all the issues. Could you please review it again and let me know if any further changes are needed? Thanks!

carolinan commented:
> The paragraphs beneath "Free", "Single" and "Expert" should not have line height defined locally, and should be set to Medium font size instead

The font sizes are set to large and needs to be updated.

>The Columns block should have Block Spacing: Regular both vertically and horizontally.

There is no block spacing set on the columns block:
![image](https://github.com/user-attachments/assets/20495283-0057-400c-930b-31e748b09202)


Why was `"margin":{"top":"0","bottom":"0"},` removed from the wrapping group block?

noruzzamans commented:
> > The paragraphs beneath "Free", "Single" and "Expert" should not have line height defined locally, and should be set to Medium font size instead
> 
> The font sizes are set to large and needs to be updated.
> 
> > The Columns block should have Block Spacing: Regular both vertically and horizontally.
> 
> There is no block spacing set on the columns block: ![image](https://private-user-images.githubusercontent.com/7422055/366704388-20495283-0057-400c-930b-31e748b09202.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTQxMTgsIm5iZiI6MTcyNjExMzgxOCwicGF0aCI6Ii83NDIyMDU1LzM2NjcwNDM4OC0yMDQ5NTI4My0wMDU3LTQwMGMtOTMwYi0zMWU3NDhiMDkyMDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTJUMDQwMzM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGVmYzY0ZTVjZTA5ZDNkZjAxYzNjMzNlMTk4NjQxOGQwN2E0MWY0ZDhiYTY4YjQ4NzgyMTU5NGJmNmJjZjBlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.W3nxtERhgx694i_ZfRdkf17xXHwbu2vnO4zsVq7XH4I)
> 
> Why was `"margin":{"top":"0","bottom":"0"},` removed from the wrapping group block?

@carolinan I have set the font sizes to large, adjusted the recommended block spacing, and set the margin in the group block to "margin":{"top":"0","bottom":"0"}. Thanks

carolinan commented on patterns/services-three-column-boxed-pricing-table.php:
```diff
@@ -0,0 +1,112 @@
+ */
+
+?>
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|60","bottom":"var:preset|spacing|60","left":"var:preset|spacing|50","right":"var:preset|spacing|50"},"blockGap":"var:preset|spacing|50","margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
```
The blockGap / block spacing is meant to be on the columns block.
To adjust the space between the columns.

noruzzamans replied:
@carolinan Apologies, I couldn't understand this earlier, but I have now resolved the issue.

carolinan commented:
> The Columns block should have Block Spacing: Regular both vertically and horizontally.

I think it is meant to be on all sides, not only top and left But if you want we can wait for feedback from @beafialho.

beafialho commented:
Unfortunately, I'm not seeing the changes I suggested, I'm seeing different ones instead.

For the sake of clarity, here's what I meant so you can compare with your current markup:

```
<!-- wp:group {"metadata":{"name":"3-column boxed pricing table"},"align":"full","className":"is-style-default","style":{"spacing":{"padding":{"top":"var:preset|spacing|60","bottom":"var:preset|spacing|60"}}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group alignfull is-style-default" style="padding-top:var(--wp--preset--spacing--60);padding-bottom:var(--wp--preset--spacing--60)"><!-- wp:group {"align":"wide","layout":{"type":"flex","flexWrap":"wrap","justifyContent":"space-between"}} -->
<div class="wp-block-group alignwide"><!-- wp:heading {"fontSize":"x-large"} -->
<h2 class="wp-block-heading has-x-large-font-size">Choose your membership</h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"className":"is-style-pill"} -->
<p class="is-style-pill">Become a member</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:columns {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|50","bottom":"0"},"blockGap":{"top":"var:preset|spacing|50","left":"var:preset|spacing|50"}}}} -->
<div class="wp-block-columns alignwide" style="margin-top:var(--wp--preset--spacing--50);margin-bottom:0"><!-- wp:column {"style":{"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}},"border":{"width":"1px","color":"#11111133","radius":"10px"}}} -->
<div class="wp-block-column has-border-color" style="border-color:#11111133;border-width:1px;border-radius:10px;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"dimensions":{"minHeight":"100%"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch","verticalAlignment":"top"}} -->
<div class="wp-block-group" style="min-height:100%"><!-- wp:group {"style":{"spacing":{"padding":{"bottom":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","verticalAlignment":"top","justifyContent":"space-between"}} -->
<div class="wp-block-group" style="padding-bottom:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"},"layout":{"selfStretch":"fixed","flexSize":"65%"}},"layout":{"type":"flex","orientation":"vertical"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">Free</h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"fontSize":"medium"} -->
<p class="has-medium-font-size">You get access to our free articles and weekly newsletter.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"right"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3,"style":{"typography":{"textDecoration":"line-through"}}} -->
<h3 class="wp-block-heading" style="text-decoration:line-through">0€</h3>
<!-- /wp:heading --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"left"}} -->
<div class="wp-block-buttons"><!-- wp:button {"width":100} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100"><a class="wp-block-button__link wp-element-button">Join</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:column -->

<!-- wp:column {"style":{"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}},"border":{"width":"1px","color":"#11111133","radius":"10px"}}} -->
<div class="wp-block-column has-border-color" style="border-color:#11111133;border-width:1px;border-radius:10px;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"dimensions":{"minHeight":"100%"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch","verticalAlignment":"top"}} -->
<div class="wp-block-group" style="min-height:100%"><!-- wp:group {"style":{"spacing":{"padding":{"bottom":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","verticalAlignment":"top","justifyContent":"space-between"}} -->
<div class="wp-block-group" style="padding-bottom:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"},"layout":{"selfStretch":"fixed","flexSize":"65%"}},"layout":{"type":"flex","orientation":"vertical"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">Single</h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"fontSize":"medium"} -->
<p class="has-medium-font-size">Get access to our paid newsletter and a limited pass for one IRL event.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"right"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">20€</h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size">Month</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"left"}} -->
<div class="wp-block-buttons"><!-- wp:button {"width":100} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100"><a class="wp-block-button__link wp-element-button">Join</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:column -->

<!-- wp:column {"style":{"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}},"border":{"width":"1px","color":"#11111133","radius":"10px"}}} -->
<div class="wp-block-column has-border-color" style="border-color:#11111133;border-width:1px;border-radius:10px;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"dimensions":{"minHeight":"100%"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch","verticalAlignment":"top"}} -->
<div class="wp-block-group" style="min-height:100%"><!-- wp:group {"style":{"spacing":{"padding":{"bottom":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","verticalAlignment":"top","justifyContent":"space-between"}} -->
<div class="wp-block-group" style="padding-bottom:var(--wp--preset--spacing--40)"><!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"},"layout":{"selfStretch":"fixed","flexSize":"65%"}},"layout":{"type":"flex","orientation":"vertical"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">Expert</h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"fontSize":"medium"} -->
<p class="has-medium-font-size">Get access to our paid newsletter and an unlimited event pass.</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|20"}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"right"}} -->
<div class="wp-block-group"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">40€</h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size">Month</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"left"}} -->
<div class="wp-block-buttons"><!-- wp:button {"width":100} -->
<div class="wp-block-button has-custom-width wp-block-button__width-100"><a class="wp-block-button__link wp-element-button">Join</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div>
<!-- /wp:group -->
```

noruzzamans commented:
@beafialho I have fixed all issues with my latest file that I pushed, and @carolinan has reviewed my changes. I think she only reviewed this part: 'The Columns block should have Block Spacing: Regular both vertically and horizontally,' which you mentioned previously. I have followed your instructions and updated it. Could you please review my latest code? Thanks.

```php
<!-- wp:columns {"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"0"},"blockGap":{"top":"var:preset|spacing|50","left":"var:preset|spacing|50"}}}} -->
```

beafialho commented:
@noruzzamans I'm still not seeing these things addressed:

> The paragraphs beneath "Free", "Single" and "Expert" should not have line height defined locally, and should be set to Medium font size instead

And it looks like "Free", "Single" and "Expert" are paragraphs now. Should they be H3?

![Captura de ecrã 2024-09-13, às 16 28 28](https://github.com/user-attachments/assets/35175618-1f42-4cf1-a1d9-cb782955c6a4)

noruzzamans commented:
> @noruzzamans I'm still not seeing these things addressed:
> 
> > The paragraphs beneath "Free", "Single" and "Expert" should not have line height defined locally, and should be set to Medium font size instead
> 
> And it looks like "Free", "Single" and "Expert" are paragraphs now. Should they be H3?
> 
> ![Captura de ecrã 2024-09-13, às 16 28 28](https://private-user-images.githubusercontent.com/65220155/367355790-35175618-1f42-4cf1-a1d9-cb782955c6a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNDU3ODMsIm5iZiI6MTcyNjI0NTQ4MywicGF0aCI6Ii82NTIyMDE1NS8zNjczNTU3OTAtMzUxNzU2MTgtMWY0Mi00Y2YxLWExZDktY2I3ODI5NTVjNmE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEzVDE2MzgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlZTY0YjU3ZDU2ZTJhZDVmNTNjZGViMjFhYzFmOGIzZGEzODdmZGNmMjYzZjU0ODY1MDViYWViMGVjYjJhYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kACyn6XT8_qHf-xcObBcp5r2EjSqTNXON2_Y14aF2lY)

@beafialho I have updated the paragraphs to H3. Could you please review them again? Thanks.

juanfra reviewed (approved):
Thank you for your contribution and your persistence on this @noruzzamans 

Looks good to me ✅

beafialho commented:
> @beafialho I have updated the paragraphs to H3. Could you please review them again? Thanks.

Thank you for your persistence, it's looking good.
